lib/spack/spack/environment/environment.py

```diff
@@ -384,7 +384,8 @@ def __init__(
         link_type="symlink",
     ):
         self.base = base_path
-        self.root = spack.util.path.canonicalize_path(root)
+        self._root = root
```

Comment:
I would change this variable name to `self.raw_root` so that we don't have to have a reference from another class to an "private" (underscore) variable in this one.